### PR TITLE
Adding new resource type to SecurityInsights - already reviewed/approved in private repo

### PR DESCRIPTION
**Pull request by matthew-ouellette**
Was review/approved for ARM and swagger already in private repo: Azure/azure-rest-api-specs-pr#3705

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-05-01-preview/examples/metadata/DeleteMetadata.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "parameters": {
+    "api-version": "2021-03-01-preview",
```
Comment (filizt) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-05-01-preview/examples/metadata/DeleteMetadata.json]: Please update with new api version

Comment (mpo-sec) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-05-01-preview/examples/metadata/DeleteMetadata.json]: Sorry I think this comment was on one of the earlier commits.  The plan for this API was to add it to `2021-03-01-preview`, and they are all under that folder etc

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-05-01-preview/Metadata.json [hunk 2]
```diff
@@ -0,0 +1,777 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/1.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
```
Comment (filizt) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-05-01-preview/Metadata.json]: Is this a long running operation? If so, it should have the x-ms-long-running-operation annotation. [http://azure.github.io/autorest/extensions/#x-ms-long-running-operation](http://azure.github.io/autorest/extensions/#x-ms-long-running-operation)

Comment (mpo-sec) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-05-01-preview/Metadata.json]: This is not long running, the put/patch should be instantaneous

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-05-01-preview/examples/metadata/DeleteMetadata.json (hunk 1 above)
Comment (filizt) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-05-01-preview/examples/metadata/DeleteMetadata.json]: The APIs and the models defined in the swagger look good. But you need to release these new path additions in a new API version. This is a process enforced by the API breaking change review board. That is also the reason why breaking changes task is failing on this RP, See more here: http://aka.ms/bcforapi

**Review by JeffreyRichter — APPROVED** [review]

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/Metadata.json [hunk 3]
```diff
@@ -0,0 +1,777 @@
...
+    "metadataProperties": {
+      "description": "Metadata property bag.",
+      "required": [
+        "id",
+        "parentId",
+        "kind"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
```
Comment (RamyasreeChakka) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/Metadata.json:603]: Top-level properties like 'id', 'kind' should be repeated under properties bag. Please rename the properties. For example: consider renaming 'id' to 'metadataId'.

Comment (RamyasreeChakka) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/Metadata.json:603]: Please refer rpc here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

**Comment by RamyasreeChakka** [comment]
@mpo-sec Please take a look at the comments.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/Metadata.json (hunk 3 above)
Comment (mpo-sec) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/Metadata.json:603]: Okay I can make this change

**Comment by mpo-sec** [comment]
@RamyasreeChakka I have updated the id fields to follow your suggestions.  Also I double checked and I have all "kind" fields and not any "type", matching the RPC

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/Metadata.json [hunk 4]
```diff
@@ -0,0 +1,777 @@
...
+        },
+        "parentId": {
+          "$ref": "#/definitions/metadataParentId",
+          "description": "Full parent resource ID of the content item the metadata is for.  This is the full resource ID including the scope (subscription and resource group)"
+        },
+        "version": {
+          "$ref": "#/definitions/metadataVersion",
+          "description": "Version of the content.  Default and recommended format is numeric (e.g. 1, 1.0, 1.0.0, 1.0.0.0), following ARM template best practices.  Can also be any string, but then we cannot guarantee any version checks"
+        },
+        "kind": {
```
Comment (RamyasreeChakka) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/Metadata.json:615]: I would recommend to rename 'kind' field as well. Kind is a top-level property in Azure. 
Look at the list of top-level properties here [https://github.com/Azure/azure-rest-api-specs/blob/119d0a9b2e51ce8b6540bf731512229e1a983069/specification/common-types/resource-management/v2/types.json#L91](https://github.com/Azure/azure-rest-api-specs/blob/119d0a9b2e51ce8b6540bf731512229e1a983069/specification/common-types/resource-management/v2/types.json#L91)

Comment (RamyasreeChakka) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-03-01-preview/Metadata.json:615]: Just rename the 'kind' property here. Rest all places should be fine.

**Comment by RamyasreeChakka** [comment]
> @RamyasreeChakka I have updated the id fields to follow your suggestions. Also I double checked and I have all "kind" fields and not any "type", matching the RPC

RPC mentioned some set of properties. You can refer the full list of top level properties here https://github.com/Azure/azure-rest-api-specs/blob/119d0a9b2e51ce8b6540bf731512229e1a983069/specification/common-types/resource-management/v2/types.json#L91. Please just rename the properties.kind and rest should be fine.

I am signing off assuming you will make above change.

**Review by RamyasreeChakka — APPROVED** [review]

**Review by njuCZ — APPROVED** [review]

**Pull request merged**